core/src/test/java/org/opensearch/sql/common/utils/StringUtilsTest.java

```diff
@@ -0,0 +1,29 @@
+package org.opensearch.sql.common.utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.opensearch.sql.common.utils.StringUtils.unquoteText;
+
+import org.junit.jupiter.api.Test;
+
+class StringUtilsTest {
+  @Test
+  void unquoteTest() {
+    assertEquals("test\\'s", unquoteText("'test\\\\'s'"));
+    assertEquals("test\\'s", unquoteText("'test\\\\\\'s'"));
```

Comment:
shouldn't this throw an error?

Reply:
Unsure... so what is happening is java is reducing the four `\` into two, and the six `\` into three. 
Once it's inside the function, it's a string that has `'test\\'s'`, which it converts into `test's`, no obligation for there to be an odd number of escape characters because it doesn't need to use them at least..
Once it's inside the function, it's a string that has `'test\\\'s'`, which it converts into `test\'s`.

I think it's behaving appropriately.

Reply:
It seems like `'` by itself won't throw an error.  Maybe this is also because the parser will throw a syntax error before ever reaching this staging in the code... so practically speaking, this is fine. 
But... then this should be properly described in the method documentation.  Otherwise, this function may be misused.